### PR TITLE
api: raise an error if the socket path is too long

### PR DESCRIPTION
**Pull request by giuseppe**
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: api.c [hunk 1]
```diff
@@ -21,7 +21,11 @@ int api_bindlisten(const char *api_socket)
     }
     memset(&addr, 0, sizeof(addr));
     addr.sun_family = AF_UNIX;
-    strncpy(addr.sun_path, api_socket, sizeof(addr.sun_path) - 1);
+    if (strlen (api_socket) >= sizeof(addr.sun_path)) {
+        fprintf(stderr, "the specified API socket path is too long (> %d)\n", sizeof(addr.sun_path));
+        return -1;
+    }
+    strcpy(addr.sun_path, api_socket);
```
Comment (AkihiroSuda) [inline api.c]: I think we should use strncpy everytime to avoid potential lint false-positive

Comment (AkihiroSuda) [reply api.c]: also make sure to terminate with NUL

Comment (giuseppe) [reply api.c]: moved back to strncpy.

The `addr.sun_path` string is NUL terminated since we have `memset(&addr, 0, sizeof(addr));`

File: api.c [hunk 2]
```diff
@@ -21,6 +21,11 @@ int api_bindlisten(const char *api_socket)
     }
     memset(&addr, 0, sizeof(addr));
     addr.sun_family = AF_UNIX;
+    if (strlen(api_socket) >= sizeof(addr.sun_path)) {
+        fprintf(stderr, "the specified API socket path is too long (> %lu)\n",
```
Comment (AkihiroSuda) [inline api.c]: nit: >=

**Pull request merged**